benchmarks/union_find/union_find.cpp

```diff
@@ -0,0 +1,179 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX_DetailsSortUtils.hpp>
+#include <ArborX_DetailsUnionFind.hpp>
+#include <ArborX_DetailsUtils.hpp>
+#include <ArborX_Version.hpp>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+#include <Kokkos_Timer.hpp>
+
+#include <boost/program_options.hpp>
+
+#include <benchmark/benchmark.h>
+
+struct UnweightedEdge
+{
+  unsigned int source;
+  unsigned int target;
+};
+
+struct AllowLoops
+{};
+struct DisallowLoops
+{};
+
+template <typename ExecutionSpace>
+Kokkos::View<UnweightedEdge *, typename ExecutionSpace::memory_space>
+buildEdges(AllowLoops, ExecutionSpace const &exec_space, int num_edges)
+{
+  using MemorySpace = typename ExecutionSpace::memory_space;
+  Kokkos::View<UnweightedEdge *, MemorySpace> edges(
+      Kokkos::view_alloc(exec_space, Kokkos::WithoutInitializing,
+                         "ArborX::Benchmark::edges"),
+      num_edges);
+
+  Kokkos::Random_XorShift1024_Pool<ExecutionSpace> rand_pool(1984);
+  Kokkos::parallel_for(
+      "ArborX::Bechmark::init_edges",
+      Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, num_edges),
+      KOKKOS_LAMBDA(unsigned i) {
+        auto rand_gen = rand_pool.get_state();
+        do
+        {
+          edges(i) = {rand_gen.urand() % num_edges,
+                      rand_gen.urand() % num_edges};
+        } while (edges(i).source == edges(i).target); // no self loops
+        rand_pool.free_state(rand_gen);
+      });
+  return edges;
+}
+
+template <typename ExecutionSpace>
+Kokkos::View<UnweightedEdge *, typename ExecutionSpace::memory_space>
+buildEdges(DisallowLoops, ExecutionSpace const &exec_space, int num_edges)
+{
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  Kokkos::Random_XorShift1024_Pool<ExecutionSpace> rand_pool(1984);
+  // Construct random permutation by sorting a vector of random values
+  Kokkos::View<int *, MemorySpace> random_values(
+      Kokkos::view_alloc(exec_space, Kokkos::WithoutInitializing,
+                         "ArborX::Benchmark::random_values"),
+      num_edges);
+  Kokkos::parallel_for(
+      "ArborX::Bechmark::init_random_values",
```

Comment:
```suggestion
      "ArborX::Benchmark::init_random_values",
```
sorry haven't noticed before